chia/run.sh

```diff
@@ -1,14 +1,20 @@
 #!/bin/bash
-#mkdir tmp 
-#for CA certificates
+
+#SSH Test if required
 if [[ "$REMOTE_LOCATION" != "local" ]]; then
-echo "Upload test before unpacking..."
+echo "SSH connection test before unpacking..."
 mkdir -p /root/.ssh/
 touch /root/.ssh/known_hosts
 echo "Found $REMOTE_HOST with user $REMOTE_USER on port $REMOTE_PORT to upload to"
-ssh-keyscan -p "${REMOTE_PORT}" "${REMOTE_HOST}" >> ~/.ssh/known_hosts
-ssh -o BatchMode=yes -o ConnectTimeout=5 -p "$REMOTE_PORT" "$REMOTE_USER"@"$REMOTE_HOST" echo Upload Destination Connection OK 2>&1 || echo "Did not pass go! Check your SSH settings" && exit 1
-else
+ssh-keyscan -p "${REMOTE_PORT}" "${REMOTE_HOST}" >> ~/.ssh/known_hosts ; export SSHPASS=${REMOTE_PASS}
+
+sshpass -e ssh -o ConnectTimeout=5 -p $REMOTE_PORT $REMOTE_USER@$REMOTE_HOST exit
+if [ $? != "0" ]; then
```

Comment:
`[ $? -ne 0 ]` is more appropriate for integers.

`[ "$?" != "0" ]` would be ok for string comparison